core/state/database_arbitrum.go

```diff
@@ -7,16 +7,11 @@ import (
 	"github.com/ethereum/go-ethereum/core/rawdb"
 )
 
-func (db *cachingDB) ActivatedAsm(moduleHash common.Hash) ([]byte, error) {
+func (db *cachingDB) ActivatedAsm(targetName string, moduleHash common.Hash) ([]byte, error) {
 	if asm, _ := db.activatedAsmCache.Get(moduleHash); len(asm) > 0 {
```

Comment:
this cache is currently broken - the cache can be populated with asms for different targets

I'll either cache whole asmMap for given moduleHash OR only use cache when targetName == local target name.